docs(router): filter the event to subscribe

santoshyadavdev opened the pull request:
The current code will not work as the `e` will be an event, 
If we try to access e.id and e.url it will throw an exception, the correct way is to filter down to the event we want to subscribe.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

jnizet commented:
That fix won't make the code more correct. What's missing is a type assertion. Something like

```
    router.events.pipe(
      filter(e => e instanceof RouterEvent),
      map(e => e as RouterEvent)
    )
```

santoshyadav198613 commented:
> That fix won't make the code more correct. What's missing is a type assertion. Something like
> 
> ```
>     router.events.pipe(
>       filter(e => e instanceof RouterEvent),
>       map(e => e as RouterEvent)
>     )
> ```

Yeah, thanks for pointing this out this will also work.

atscott commented:
How about `filter((e: Event): e is RouterEvent => e instanceof RouterEvent)` instead of adding `map`?

santoshyadav198613 commented:
> How about `filter((e: Event): e is RouterEvent => e instanceof RouterEvent)` instead of adding `map`?

The code throws 
Property 'subscribe' does not exist on type 'MonoTypeOperatorFunction<Event>'

atscott commented:
@santoshyadav198613 - Did you write it like this?

![image](https://user-images.githubusercontent.com/479713/81719552-0ddbb680-9432-11ea-94e6-0b39f3011997.png)
https://stackblitz.com/edit/angular-ivy-ryq3mv

santoshyadav198613 commented:
Ahh my bad, let me try it sometime.

santoshyadavdev committed:
docs(router): filter the event to subscribe

The current code will not work as the `e` will be an event,
If we try to access e.id and e.url it will throw an exception, the correct way is to use map or filter down to specific events

santoshyadav198613 commented:
Thanks, @atscott  it works changed and committed the code.

atscott reviewed (approved):
LGTM - thanks!

santoshyadav198613 commented:
> LGTM - thanks!

Do I need to rebase, looks like some test is breaking.

atscott commented:
@santoshyadav198613 - I just triggered a rerun of the failed test in case it was a flake. If it fails again, you can try rebasing.

santoshyadav198613 commented:
> @santoshyadav198613 - I just triggered a rerun of the failed test in case it was a flake. If it fails again, you can try rebasing.

Thanks :)
